### ISSUE TITLE
Switch to v3 AWS SDK

### ISSUE THREAD

**Issue by Cretezy** [trieb-work/strapi-provider-cloudflare-r2#10]
v2 will be EOL in 2023, v3 is now recommended.

AWS docs: https://docs.aws.amazon.com/sdk-for-javascript/v3/developer-guide/migrating-to-v3.html

Cloudflare docs: https://developers.cloudflare.com/r2/examples/aws/aws-sdk-js-v3/

I just migrated one of my projects, which basically uses the same API as this one, and here are the changes I did:
- Change `aws-sdk` to `@aws-sdk/client-s3` & `@aws-sdk/lib-storage` in `package.json`
- Use the following to upload files:
  - ```ts
    const s3 = new S3({
	  credentials: {
	    accessKeyId: process.env.CF_ACCESS_KEY_ID!,
	    secretAccessKey: process.env.CF_ACCESS_SECRET!,
	  },
	  endpoint: process.env.CF_ENDPOINT,
	  region: "auto",
    });

    await new Upload({
      client: s3,
      params: {
        Key: ...,
        Body: ...,
        ContentType: ...,
        Bucket: process.env.CF_BUCKET!,
      },
    }).done();
	```

**Comment by tilman** [comment]
Good point 👍 Are you willing to open a PR? Would be great since you already made the migration somewhere else ;)

**Comment by Cretezy** [comment]
Yes can look into it!

**Comment by anthlasserre** [comment]
Hey guys 👋🏼 
I have made the upgrade to V3. Anyone would be able to test it on their side and with a public bucket, please?

**Comment by tilman** [comment]
> Hey guys 👋🏼
> I have made the upgrade to V3. Anyone would be able to test it on their side and with a public bucket, please?

Thanks for your PR. Please see my comments on it for some feedback
